packages/web3-eth-ens/src/resolver.ts

```diff
@@ -12,8 +12,8 @@ import { namehash } from './utils';
 export class Resolver {
 	private readonly registry: Registry;
 
-	public constructor(registry: Registry) {
-		this.registry = registry;
+	public constructor(registry?: Registry) {
+		this.registry = registry ?? new Registry();
```

Comment:
We should always reuse existing registry instance for multiple resolver. For efficient memory usage. 
or is there any scenario where its unavoidable?

```suggestion
	public constructor(registry: Registry) {
		this.registry = registry;
```

Reply:
I believe we should be able to reuse registry